src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -261,8 +261,8 @@ public static String relativizeAndNormalizePath(final String baseDirectory, fina
             resultPath = path;
         }
         else {
-            final Path pathAbsolute = Paths.get(path).normalize();
-            final Path pathBase = Paths.get(baseDirectory).normalize();
+            final Path pathAbsolute = Paths.get(path);
```

Comment:
Looks like we need to test message of violation and validate how path is starting https://www.google.com/amp/s/www.geeksforgeeks.org/path-normalize-method-in-java-with-examples/amp/

To be OS agnostic we usually avoid checking full path in violation, we might do only file name

Reply:
What we can test is ... We set path with `.` and/or `..` but have in violation path without such symbols.

Reply:
I have tried multiple ways but at the i didnt find a way in which relativize path is getting affected.

Reply:
yes, such normalization is done by jdk
https://github.com/openjdk/jdk/blob/db3402577a2c14a41045753a1ffe2829a6bdda91/src/java.base/unix/classes/sun/nio/fs/UnixPath.java#L478-L495

we already have test on this https://github.com/checkstyle/checkstyle/blob/bd3a9bb9ec9cf804f628171c2e7557b0afbcfd93/src/test/java/com/puppycrawl/tools/checkstyle/utils/CommonUtilTest.java#L221-L224

Reply:
![image](https://github.com/checkstyle/checkstyle/assets/812984/5aabaf8e-e984-40da-9c03-0196b87e800d)

Reply:
> We do not need to care about path in zips 

You may want to recheck this. Some users put configuration files into external JARs controlled by Maven. Those are basically ZIP files.

Reply:
We will happily fix issue if they will show how to reproduce issue.